pkg/process/checks/container.go

```diff
@@ -155,9 +155,15 @@ func fmtContainers(ctrList []*containers.Container, lastRates map[string]util.Co
 		if err != nil {
 			log.Errorf("unable to retrieve tags for container: %s", err)
 			tags = []string{}
+			if ctr.Type == containers.RuntimeNameGarden {
+				// If there is an error retrieving tags, don't send the container for garden. It means it hasn't yet been
+				// discovered by the cluster agent, so avoid sending something with no tags, i.e. no container name, ...
+				log.Debugf("No tags found for app %s, it has probably not been discovered by the DCA, skipping.", ctr.ID)
+				continue
+			}
```

Comment:
Is there a way to throw a more generalized error that can be thrown and support more cases than just Garden containers here, or is this very specific behavior to Garden? I imagine we might have subtle init race conditions elsewhere too?

Reply:
I know this behavior happens with garden, not sure about anything else, and I wanted to have a very limited impact.

However, IMO, we shouldn't ever send container payloads with empty tags, since no tags mean not `container_name` tags, means something weird in our UI. So worth considering for everything indeed.

Whenever the tagger returns an error, we should probably skip the container regardless of the type.
As I said though, I have very little knowledge of all the rest, so up to your team I guess.

Reply:
@ahmed-mez: any thoughts on how we should deal with this generally from the perspective of `container-integrations`?